tools/build.py

```diff
@@ -0,0 +1,150 @@
+    cmd_result = subprocess.run(cmd, shell=True)
+    assert cmd_result.returncode == 0
+
+    # change current working directory
+    print(f"[SPEAR | build.py] Changing current working dir from {os.getcwd()} to {os.path.join(args.clone_dir, 'tools')}")
+    os.chdir(os.path.join(args.clone_dir, "tools"))
+
+    if sys.platform == "win32":
+        cmd = ["python", "build_third_party_libs.py", "--num_parallel_jobs", f"{args.num_parallel_jobs}"]
+    elif sys.platform in ["linux", "darwin"]:    
+        cmd = [f"python build_third_party_libs.py --num_parallel_jobs {args.num_parallel_jobs}"]
+    else:
+        assert False
+    print(f"[SPEAR | build.py] Executing: {' '.join(cmd)}")
+    cmd_result = subprocess.run(cmd, shell=True)
+    assert cmd_result.returncode == 0
+
+    if sys.platform == "win32":
+        cmd = ["conda", "activate", "spear-env&", "python", "create_symbolic_links.py"]
+    elif sys.platform in ["linux", "darwin"]:
+        cmd = [f". {args.conda_path}; conda activate spear-env; python create_symbolic_links.py"]
+    else:
+        assert False
+    print(f"[SPEAR | build.py] Executing: {' '.join(cmd)}")
+    cmd_result = subprocess.run(cmd, shell=True)
+    assert cmd_result.returncode == 0
+
+    if sys.platform == "win32":
+       cmd = ["conda", "activate", "spear-env&", "python", "generate_config.py"]
+    elif sys.platform in ["linux", "darwin"]: 
+        cmd = [f". {args.conda_path}; conda activate spear-env; python generate_config.py"]
+    else:
+        assert False
+    print(f"[SPEAR | build.py] Executing: {' '.join(cmd)}")
+    cmd_result = subprocess.run(cmd, shell=True)
+    assert cmd_result.returncode == 0
+
+    # build the executable
+    if sys.platform == "win32":
+        build_exe = os.path.join(args.unreal_engine_dir, "Engine", "Build", "BatchFiles", "RunUAT.bat")
+    elif sys.platform in ["linux", "darwin"]:
+        build_exe = os.path.join(args.unreal_engine_dir, "Engine", "Build", "BatchFiles", "RunUAT.sh")
+    else:
+        assert False
+    
+    cmd = [
+        build_exe,
+        "BuildCookRun",
+        "-project=" + os.path.join(args.clone_dir, "cpp", "unreal_projects", "SpearSim", "SpearSim.uproject"), 
+        "-build",
+        "-cook",
+        "-stage",
+        "-package",
+        "-archive",
+        "-pak",
+        "-targetplatform=" + args.target_platform,
+        "-target=SpearSim",
+        "-archivedirectory=" + args.output_dir,
+        "-clientconfig=" + args.config_mode,
+    ]
+    print(f"[SPEAR | build.py] Executing: {' '.join(cmd)}")
+    cmd_result = subprocess.run(cmd)
+    assert cmd_result.returncode == 0
+
+    # change working directory
+    print(f"[SPEAR | build.py] Changing current working dir from {os.getcwd()} to {cwd}")
+    os.chdir(cwd)
+
+    print(f"[SPEAR | build.py] Successfully built SpearSim for {args.target_platform} platform with {args.config_mode} mode.")
+
+    # zip the exectuable for distribution
```

Comment:
need to code sign on mac

Reply:
Added it as todo, I'll integrate it in this PR once https://github.com/isl-org/spear/pull/152 is merged.